cli/command/cli.go

```diff
@@ -125,25 +131,42 @@ func (cli *DockerCli) Initialize(opts *cliflags.ClientOptions) error {
 	if err != nil {
 		return err
 	}
+	hasExperimental, err := isEnabled(cli.configFile.ExperimentalCli)
+	if err != nil {
+		return err
+	}
+	cli.clientInfo = ClientInfo{
+		DefaultVersion:  cli.client.ClientVersion(),
+		HasExperimental: hasExperimental,
+	}
 	cli.initializeFromClient()
 	return nil
 }
 
-func (cli *DockerCli) initializeFromClient() {
-	cli.defaultVersion = cli.client.ClientVersion()
+func isEnabled(value string) (bool, error) {
+	switch value {
+	case "enabled":
+		return true, nil
+	case "", "disabled":
+		return false, nil
+	default:
+		return false, errors.Errorf("%q is not valid, should be either enabled or disabled", value)
```

Comment:
Think we should mention the configuration option here, currently it shows:

```
$ docker version
"Moby was here" is not valid, should be either enabled or disabled
```

Perhaps something like:

```
Invalid value for "experimentalCli": "Moby was here". Valid options are "enabled" and "disabled"
```

(better suggestions welcome)